check.go

```diff
@@ -304,3 +304,37 @@ func ErrorIs(target error) CheckFunc {
 		return nil
 	}
 }
+
+// ContainsMatch returns a check function that fails if the test value does not
+// contain the check. Accepts input of type array and slice.
+func ContainsMatch(c Check) CheckFunc {
+	return contains(c, nil)
+}
+
+// Contains returns a check function that fails if the test value does not
+// contain the input. Accepts input of type array and slice.
+func Contains(v interface{}) CheckFunc {
+	return contains(DeepEqual(v), v)
+}
+
+func contains(c Check, val interface{}) CheckFunc {
+	return func(got interface{}) error {
+		rv := reflect.ValueOf(got)
+		switch rv.Kind() {
+		case reflect.Array, reflect.Slice:
+		default:
+			return FailGot(msgNotSliceArrType, got)
+		}
+
+		for j := 0; j < rv.Len(); j++ {
+			if c.Check(Index(got, j)) == nil {
+				return nil
+			}
+		}
+
+		if val == nil {
+			return FailGot(msgContainsMatch, got)
+		}
```

Comment:
This is a corner-case, but maybe we should not check for nil; we ideally want a nice error also for `Contains(nil)`.

We could e.g. use a private type:
```go
type undefinedT uint

const undefined undefinedT = 0

func ContainsMatch(c Check) CheckFunc {
 	return contains(c, undefined)
 }

// ...

func contains(c Check, expect interface{}) CheckFunc {
    // ...
    if expect == undefined {
        return FailGot(msgContainsMatch, expect)
    }
}


```
Or add another parameter:
```go
func contains(c Check, failExpect bool, expect interface{}) CheckFunc {
    // ...
}
```

Or we solve it somehow inside ContainsMatch and drop the `contains` helper:
```go

func ContainsMatch(c Check) CheckFunc {
	// ...
	return func(got interface{}) error
		var subfail Failure
		for j := 0; j < rv.Len(); j++ {
				err: = c.Check(Index(got, j))
				switch {
				case err == nil:
					return error
				case error.As(err, &subfail):
 				}
		}
		msg := fmt.Sprtinf("check %T does not match any elements")
		fail := FailGot(msg, got)
		fail.Expect = subfail.Expect // may be empty
		return fail
}
```

Reply:
"%T" doesn't work too great for anonymous CheckFunc, so dropping that.